uat/custom-components/client-python-paho/src/grpc_client_server/grpc_control_server.py

```diff
@@ -184,3 +303,61 @@
         # TODO
         self.__logger.info("UnsubscribeRequest Placeholder TODO")
         return MqttSubscribeReply()
+
+    @staticmethod
+    def convert_conn_ack(conn_ack_info: ConnAckInfo) -> Mqtt5ConnAck:  # pylint: disable=too-many-branches

```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this function is 16. By comparison, 98% of the functions in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the function has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this function or break it into multiple functions.

Reply:
fixed in c1b9703